components/block/element.js

```diff
@@ -9,56 +9,93 @@ export default class Block extends LitElement {
   static styles = [style];
 
   static properties = {
-    blockId: { type: Number},
+    blockId: {type: String, attribute: 'data-ngl-block-id'},
   };
 
-  renderMenu(){
-    return html`
-      <nav>
-        <button @click=${this.edit}>Edit</button>
-        <button @click=${this.moveUp}>Move Up</button>
-        <button @click=${this.moveDown}>Move Down</button>
-        <button @click=${this.delete}>Delete</button>
-      </nav>
-    `;
-  }
+  constructor() {
+    super();
 
-  edit(e){
-    console.log(e.target.innerText)
-    this.refresh();
+    this.isSaveListenerAttached = false;
   }
 
-  delete(e) {
-    console.log(e.target.innerText)
-    this.remove();
+  getBlock() {
+    return window.parent.Core.g.layout.blocks.findWhere({id: this.blockId});
   }
 
-  moveUp(e) {
-    console.log(e.target.innerText)
-    const prevElement = this.previousElementSibling;
-    isBlock(prevElement) && this.parentNode.insertBefore(this, prevElement)
+  renderMenu() {
+    const block = this.getBlock();
+    const isInLinkedZone = block.zone().is_linked();
+
+    if (isInLinkedZone) {
+      return html`
+        <div class="edit-menu">
+          <button @click=${this.refresh}>Refresh</button>
+        </div>
+      `;
+    } else {
+      const parentId = block.attributes.parent_block_id;
+      const parent = document.querySelector(
+        `ngl-block[data-ngl-block-id="${parentId}"]`
+      );
+
+      if (parent && isBlock(parent)) {
+        return html`
+          <div class="edit-menu">
+            <button @click=${parent.edit.bind(parent)}>Edit container</button>
+            <button @click=${parent.refresh.bind(parent)}>
+              Refresh container
+            </button>
+            <button @click=${this.edit}>Edit</button>
+            <button @click=${this.refresh}>Refresh</button>
+          </div>
+        `;
+      } else {
+        return html`
+          <div class="edit-menu">
+            <button @click=${this.edit}>Edit</button>
+            <button @click=${this.refresh}>Refresh</button>
+          </div>
+        `;
+      }
+    }
   }
 
-  moveDown(e) {
-    console.log(e.target.innerText)
-    const nextElement = this.nextElementSibling;
-    isBlock(nextElement) && this.parentNode.insertBefore(nextElement, this)
+  edit() {
+    const block = this.getBlock();
+    block.trigger('edit');
+
+    if (!this.isSaveListenerAttached) {
+      block.on('sidebar_save:success', () => {
+        this.refresh();
+      });
+      this.isSaveListenerAttached = true;
+    }
```

Comment:
This should be configured in connectedCallback and in that case you don't need `isSaveListenerAttached`.

Read more here:
https://lit.dev/docs/components/lifecycle/#connectedcallback


```js
connectedCallback() {
  super.connectedCallback();
  this.block.on('sidebar_save:success', () => {
    this.refresh();
  });
}
```

:information_source:  connectedCallback is absolutely the best thing you get for free in native web components.

Reply:
Fixed in dbab07ec298e517830df04a0d10f714841c90098